### PR TITLE
Fix/insights load speed

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Uses skipBatch flag when calling trpc in every graph on insights
- Implements a view that should load booking time status and aggregates userId and teamId.
- Adds performance script that creates many bookings and many team users.

Fixes #

**Environment**: Staging(main branch) 

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] On a reset DB you can navigate to packages/prisma and run yarn seed-insights once.
- [ ] Go to file seed-insights and edit createExtraMembers = true
- [ ] Run yarn seed-insights again a couple times.
- [ ] Enter with insights admin credentials
- [ ] Navigate to /insights and take a look of load times.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/insights/components/AverageEventDurationChart.tsx [hunk 1]
```diff
@@ -16,13 +16,20 @@ export const AverageEventDurationChart = () => {
...
+      startDate: startDate.toISOString(),
+      endDate: endDate.toISOString(),
+      teamId: teamId ?? undefined,
+      memberUserId: selectedMemberUserId ?? undefined,
+      userId: selectedUserId ?? undefined,
+    },
+    {
+      trpc: {
+        context: { skipBatch: true },
+      },
```
Comment (alannnc) [inline packages/features/insights/components/AverageEventDurationChart.tsx:30]: With this change we tell trpc to don't batch this call.

File: packages/features/insights/server/events.ts [hunk 2]
```diff
@@ -11,10 +11,13 @@ interface ITimeRange {
...
 class EventsInsights {
-  static getBookingsInTimeRange = async (timeRange: ITimeRange, where: Prisma.BookingWhereInput) => {
+  static getBookingsInTimeRange = async (
+    timeRange: ITimeRange,
+    where: Prisma.BookingTimeStatusWhereInput
+  ) => {
     const { start, end } = timeRange;
 
-    const events = await prisma.booking.count({
+    const events = await prisma.bookingTimeStatus.count({
```
Comment (alannnc) [inline packages/features/insights/server/events.ts:20]: Using our new performance view

File: packages/prisma/schema.prisma [hunk 3]
```diff
@@ -8,7 +8,7 @@ datasource db {
 
 generator client {
   provider        = "prisma-client-js"
-  previewFeatures = []
+  previewFeatures = ["views"]
```
Comment (alannnc) [inline packages/prisma/schema.prisma:11]: required for prisma preview

File: packages/prisma/seed-insights.ts [hunk 4]
```diff
@@ -392,3 +392,149 @@ main()
     await prisma.$disconnect();
     process.exit(1);
   });
+
+/**
+ * This will create many users in insights teams with bookings 1y in the past
+ * Should be run after the main function is executed once
+ */
+async function createPerformanceData() {
```
Comment (alannnc) [inline packages/prisma/seed-insights.ts:400]: This script should create 50 or more team members with several bookings.

File: packages/features/insights/context/FiltersProvider.tsx [hunk 5]
```diff
@@ -117,18 +117,21 @@ export function FiltersProvider({ children }: { children: React.ReactNode }) {
         setSelectedTimeView: (selectedTimeView) => setSelectedTimeView(selectedTimeView),
         setSelectedMemberUserId: (selectedMemberUserId) => {
           setSelectedMemberUserId(selectedMemberUserId);
+          const { userId, eventTypeId, ...rest } = router.query;
```
Comment (alannnc) [inline packages/features/insights/context/FiltersProvider.tsx:120]: Fixes some scenarios that should remove queries from url.

Comment (sean-brydon) [reply packages/features/insights/context/FiltersProvider.tsx:120]: Would useTypedQuery help here? Essentially does this but with some helper functions

**Review by sean-brydon — COMMENTED** [review]
LGTM - tested and its much better without batch mode on these queries 

Will wait for a second eye tho

File: packages/features/insights/context/FiltersProvider.tsx (hunk 5 above)
Comment (zomars) [reply packages/features/insights/context/FiltersProvider.tsx:120]: useTypedQuery can't handle multiple changes from one call. I've tried and it was very flaky.

**Review by zomars — COMMENTED** [review]
Code LGTM. Testing locally before approval.

**Comment by PeerRich** [comment]
getting this really weird behavior: https://app.birdeatsbug.com/sessions/l0toVe27aduB1BqFZjPfxaMV7C67BaFuBwFNpgh4CuHL

not sure if this is due to staging DB.

whats the best way to test this PR with production data?

**Review by zomars — APPROVED** [review]
Added stale time to 30 seconds to prevent revalidating too often.

Ship it

**Pull request merged**